Game/Entities/Player.cs

```diff
@@ -31,6 +52,132 @@ public override void _PhysicsProcess(double delta)
 			velocity.Y = Mathf.MoveToward(Velocity.Y, 0, Speed);
 		}
 
+		// Chek that the player is not performing a one-off action
+		string animation = animPlayer.CurrentAnimation;
+		foreach (string anim in animActions) 
+		{
+			if (anim == animation) 
+			{
+				running = false;
+				break;
+			} 
+			else 
+			{
+				running = true;
+			}
+		}
+		
+		// Start running or idle animations
+		if(running) {
+			if (direction != Vector2.Zero)
+			{
+				if (direction.X > 0) 
+				{
+					if (direction.Y > 0)
+					{
+						animPlayer.Play("Down-Right");
+						prevDirection = Vector2.Right + Vector2.Down;
+					} 
+					else if (direction.Y < 0)
+					{
+						animPlayer.Play("Up-Right");
+						prevDirection = Vector2.Right + Vector2.Up;
+					}
+					else
+					{
+						animPlayer.Play("Right");
+						prevDirection = Vector2.Right;
+					}
+				} 
+				else if (direction.X < 0)
+				{
+					
+					if (direction.Y > 0)
+					{
+						animPlayer.Play("Down-Left");
+						prevDirection = Vector2.Left + Vector2.Down;
+					} 
+					else if (direction.Y < 0)
+					{
+						animPlayer.Play("Up-Left");
+						prevDirection = Vector2.Left + Vector2.Up;
+					}
+					else
+					{
+						animPlayer.Play("Left");
+						prevDirection = Vector2.Left;
+					}
+				}
+				else 
+				{
+					if (direction.Y > 0)
+					{
+						animPlayer.Play("Down");
+						prevDirection = Vector2.Down;
+					} 
+					else if (direction.Y < 0)
+					{
+						animPlayer.Play("Up");
+						prevDirection = Vector2.Up;
+					}
+				}
+			} 
+			else if (prevDirection != Vector2.Zero)
+			{
+				if (prevDirection.X > 0) 
+				{
+					if (prevDirection.Y > 0)
+					{
+						animPlayer.Play("Idle Down-Right");
+					} 
+					else if (prevDirection.Y < 0)
+					{
+						animPlayer.Play("Idle Up-Right");
+					}
+					else
+					{
+						animPlayer.Play("Idle Right");
+					}
+				} 
+				else if (prevDirection.X < 0)
+				{
+					
+					if (prevDirection.Y > 0)
+					{
+						animPlayer.Play("Idle Down-Left");
+					} 
+					else if (prevDirection.Y < 0)
+					{
+						animPlayer.Play("Idle Up-Left");
+					}
+					else
+					{
+						animPlayer.Play("Idle Left");
+					}
+				}
+				else 
+				{
+					if (prevDirection.Y > 0)
+					{
+						animPlayer.Play("Idle Down");
+					} 
+					else if (prevDirection.Y < 0)
+					{
+						animPlayer.Play("Idle Up");
+					}	
+				}
+
+			}
+			else 
+			{
+				if (animPlayer.CurrentAnimation != "Die" && animPlayer.IsPlaying() == false) 
+				{
+					animPlayer.Play("Idle Down");
+				}
+			}
+
+		}
+		
```

Comment:
Improve this logic as discussed, maybe with AnimationTree?

Reply:
Fixed with some angle integer math and renaming animations with indicies.